src/test/java/com/agnieszkapawska/flashcards/domain/facades/FlashcardFacadeTest.java

```diff
@@ -0,0 +1,111 @@
+package com.agnieszkapawska.flashcards.domain.facades;
+
+import com.agnieszkapawska.flashcards.FlashcardAndQuestionTagAbstractTests;
+import com.agnieszkapawska.flashcards.domain.dtos.FlashcardSaveDto;
+import com.agnieszkapawska.flashcards.domain.dtos.FlashcardSaveResponseDto;
+import com.agnieszkapawska.flashcards.domain.models.Flashcard;
+import com.agnieszkapawska.flashcards.domain.models.QuestionTag;
+import com.agnieszkapawska.flashcards.domain.services.FlashcardService;
+import com.agnieszkapawska.flashcards.domain.services.QuestionTagService;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import java.util.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+public class FlashcardFacadeTest extends FlashcardAndQuestionTagAbstractTests {
+    @Autowired
+    private FlashcardFacade flashcardFacade;
+    @MockBean
+    private FlashcardService flashcardService;
+    @MockBean
+    private QuestionTagService questionTagService;
+    private Flashcard flashcard;
+
+    @Before
+    public void createFlashcard() {
+        flashcard = new Flashcard();
+        flashcard.setId(1L);
+        flashcard.setQuestion("question");
+        flashcard.setAnswer("answer");
+        flashcard.setExplanation("explanation");
+        //questionTag with flashcards
+        QuestionTag questionTag = new QuestionTag();
+        questionTag.setName("home");
+        questionTag.setFlashcards(new HashSet<>(Collections.singletonList(new Flashcard("kubek", "mug"))));
```

Comment:
Woudln't it be better to use here flashcard created few lines above?

Reply:
I want to have flashcard with question tag which contains another flashcard